dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNHvdcLineInterfaceIIDM.cpp

```diff
@@ -74,12 +74,30 @@ void HvdcLineInterfaceIIDM::exportStateVariablesUnitComponent() {
     case ConverterInterface::VSC_CONVERTER: {
       shared_ptr<VscConverterInterfaceIIDM> vsc1 = dynamic_pointer_cast<VscConverterInterfaceIIDM>(conv1_);
       shared_ptr<VscConverterInterfaceIIDM> vsc2 = dynamic_pointer_cast<VscConverterInterfaceIIDM>(conv2_);
-      (vsc1->getVscIIDM()).getTerminal().setP(-1 * getValue<double>(VAR_P1) * SNREF);
-      (vsc1->getVscIIDM()).getTerminal().setQ(-1 * getValue<double>(VAR_Q1) * SNREF);
-      (vsc2->getVscIIDM()).getTerminal().setP(-1 * getValue<double>(VAR_P2) * SNREF);
-      (vsc2->getVscIIDM()).getTerminal().setQ(-1 * getValue<double>(VAR_Q2) * SNREF);
       bool connected1 = (getValue<int>(VAR_STATE1) == CLOSED);
       bool connected2 = (getValue<int>(VAR_STATE2) == CLOSED);
+      if (connected1) {
```

Comment:
It means that if disconnected and the value is not 0 then the assigment is not done